App.tsx

```diff
@@ -309,11 +314,66 @@ const App = () => {
               color="#841584"
             />
             <Text style={styles.debug}>{JSON.stringify(travelTime)}</Text>
-            <Button
-              onPress={() => onShare(destinationTextSaved)}
-              title="Share"
-              color="#841584"
-            />
+            {travelTime.walking !== undefined ? (
+              <Button
+                onPress={() =>
+                  onShare(
+                    `I need ${formatTravelTime(
+                      JSON.stringify(travelTime.walking)
+                    )} to walk to ${destinationTextSaved}.`
+                  )
+                }
+                title={`🚶 ${formatTravelTime(
+                  JSON.stringify(travelTime.walking)
+                )}`}
+                color="#841584"
+              />
+            ) : null}
+            {travelTime.bicycling !== undefined ? (
+              <Button
```

Comment:
This sequence repeats itself. Repetition is evil. I think this is a good fit for a custom component (https://reactnative.dev/docs/intro-react#custom-components), like the `GoogleMapsImage`.

Reply:
For example:
```jsx
<TravelTimeShareButton destination='{destinationTextSaved}' time='{travelTime.walking}' mode='walking' />
```

Reply:
Inside the ternary operator of course.